src/main/java/org/broadinstitute/consent/http/db/mapper/DacWithDaasReducer.java

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import java.util.Map;
+import java.util.Objects;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.mapper.MappingException;
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+public class DacWithDaasReducer implements LinkedHashMapRowReducer<Integer, Dac>,
+    ConsentLogger {
+
+
+  @Override
+  public void accumulate(Map<Integer, Dac> container, RowView rowView) {
+
+    try {
+
+      Dac dac =
+          container.computeIfAbsent(
+              rowView.getColumn("dac_id", Integer.class), id -> rowView.getRow(Dac.class));
+
+      if (Objects.nonNull(rowView.getColumn("daa_daa_id", Integer.class))) {
+        DataAccessAgreement daa = new DataAccessAgreement();
+
+        try {
+          if (dac != null && rowView.getColumn("daa_daa_id", Integer.class) != null) {
+            daa = rowView.getRow(DataAccessAgreement.class);
+          }
+        } catch (MappingException e) {
```

Comment:
We should at least log these at the warn level so we can follow up. I see in other reducer class cases we're logging messages, but we should start logging the full stack trace to make debugging easier.

Reply:
I've noticed now, when logging these at the warn level, the logs are getting _polluted_ with these warning messages when the getById endpoint is called. @rushtong I'm not sure if this is what we want...

https://github.com/DataBiosphere/consent/assets/68303030/490b74d9-336f-41e8-971a-dc630132c7f1

Reply:
I think I know what that is ... that's from this:
```
rowView.getColumn("daa_daa_id", Integer.class) != null
```
If there is no value in column `daa_daa_id`, then the value that comes back is `0`. What you need to do is check to see if it's also `> 0`. That's because jdbc casts null to `0` since it is a primitive `int` and can't be a proper `null`

Reply:
Ah, very interesting. That seemed to do the trick— thank you!